include/validator/validator_component.h

```diff
@@ -142,20 +167,61 @@ struct InstanceExprVisitor {
         return Unexpect(ErrCode::Value::MissingArgument);
       }
 
-      if (!matchImportAndArgTypes(ImportDesc, ArgIt->getIndex().getSort())) {
+      const auto &ArgSortIndex = ArgIt->getIndex();
+      const auto &ArgSort = ArgSortIndex.getSort();
+      const uint32_t ArgIdx = ArgSortIndex.getSortIdx();
+
+      if (!matchImportAndArgTypes(ImportDesc, ArgSort)) {
         spdlog::error(ErrCode::Value::ArgTypeMismatch);
         spdlog::error("Component[{}]: Argument '{}' type mismatch"sv,
                       Inst.getComponentIdx(), ImportName);
         return Unexpect(ErrCode::Value::ArgTypeMismatch);
       }
+
+      if (!isValidIndex(ArgSort, ArgIdx)) {
```

Comment:
`isInvalid?Index` and the `?` should be replace with a proper noun, e.g. `isInvalidArgumentIndex`